cmd/tenderdash/main.go

```diff
@@ -64,3 +67,30 @@ func main() {
 		os.Exit(2)
 	}
 }
+
+func newLoggerFromConfig(conf *config.Config) (log.Logger, func(), error) {
+	var (
+		writer    io.Writer = os.Stderr
+		closeFunc           = func() {}
+		err       error
+	)
+	if conf.LogFilePath != "" {
+		file, err := os.OpenFile(conf.LogFilePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create log writer: %w", err)
+		}
+		closeFunc = func() {
+			_ = file.Close()
+		}
+		writer = io.MultiWriter(writer, file)
```

Comment:
I just wondering how it works. Does it flush every line into file or it's optimized for logging and has some buffer and flushing logic to do not introduce too much overhead?

Reply:
this is a documentation
```
// MultiWriter creates a writer that duplicates its writes to all the
// provided writers, similar to the Unix tee(1) command.
//
// Each write is written to each listed writer, one at a time.
// If a listed writer returns an error, that overall write operation
// stops and returns the error; it does not continue down the list.
```

Reply:
Yeah, but does it have some buffer and flush logic? @lklimek

Reply:
no, it doesn't use buffer